integrations/sentry/lib/index.js

```diff
@@ -7,57 +7,100 @@
+  .option('maxMessageLength', null) // deprecated
   .option('logger', null)
   .option('customVersionProperty', null)
+  .option('debug', false)
+  .tag(
+    'sentry',
+    '<script src="https://browser.sentry-cdn.com/5.12.1/bundle.min.js" integrity="sha384-y+an4eARFKvjzOivf/Z7JtMJhaN6b+lLQ5oFbBbUwZNNVir39cYtkjW1r6Xjbxg3" crossorigin="anonymous"></script>'
+  )
+  // Sentry.Integrations.RewriteFrames plugin: https://docs.sentry.io/platforms/javascript/#rewriteframes
   .tag(
-    '<script src="https://cdn.ravenjs.com/3.17.0/raven.min.js" crossorigin="anonymous">'
+    'plugin',
+    '<script src="https://browser.sentry-cdn.com/5.12.1/rewriteframes.min.js" crossorigin="anonymous"></script>'
   ));
 
 /**
  * Initialize.
+ * https://docs.sentry.io/error-reporting/quickstart/?platform=browser
  *
- * https://docs.sentry.io/clients/javascript/config/
- * https://github.com/getsentry/raven-js/blob/3.12.1/src/raven.js#L646-L649
  * @api public
  */
 
 Sentry.prototype.initialize = function() {
-  var dsnPublic = this.options.config;
   var customRelease = this.options.customVersionProperty
     ? window[this.options.customVersionProperty]
     : null;
-  var options = {
-    logger: this.options.logger,
+
+  var config = {
+    dsn: this.options.config,
+    environment: this.options.environment,
     release: customRelease || this.options.release,
     serverName: this.options.serverName,
     whitelistUrls: this.options.whitelistUrls,
+    blacklistUrls: this.options.ignoreUrls,
+    // ignoreErrors still exists, but is not documented on Sentry's website
+    // https://github.com/getsentry/sentry-javascript/blob/master/packages/core/src/integrations/inboundfilters.ts#L12
     ignoreErrors: this.options.ignoreErrors,
-    ignoreUrls: this.options.ignoreUrls,
-    includePaths: this.options.includePaths,
-    maxMessageLength: this.options.maxMessageLength
+    integrations: [],
+    debug: this.options.debug
   };
 
-  window.RavenConfig = {
-    dsn: dsnPublic,
-    config: reject(options)
-  };
+  var logger = this.options.logger;
+  var includePaths = this.options.includePaths;
+
+  var self = this;
+  this.load('sentry', function() {
+    self.load('plugin', function() {
+      // values from `includePaths` tells the Sentry app which frames in a StackTrace to display in the user's
+      // dashboard. we use the Sentry.Integrations.RewriteFrames plugin to check each frame in a stacktrace and reassign
+      // the value of frame.in_app to true/false depending on whether we find a match: https://docs.sentry.io/platforms/javascript/#rewriteframes
+      if (includePaths.length > 0) {
+        config.integrations.push(
+          new window.Sentry.Integrations.RewriteFrames({
+            iteratee: function(frame) {
+              for (var i = 0; i < includePaths.length; i++) {
+                try {
+                  if (frame.filename.match(new RegExp(includePaths[i]))) {
```

Comment:
Do you know how often `iteratee` function is run?

`new RegExp` has performance implications. I'm thinking you could run `new RegExp` outside of the `iteratee` function just once per `includePath` for better performance.

Reply:
Yeah, excellent point - I can def run new `RegExp` outside the iteratee function. Thanks!